velox/vector/VariantToVector.cpp

```diff
@@ -89,6 +111,12 @@ ArrayVectorPtr variantArrayToVector(
   }
 
   const auto elementType = variantArray.at(nonNullIndex.value()).inferType();
+  if (elementType->kind() == TypeKind::SHORT_DECIMAL) {
```

Comment:
Initialize Type here. Remove specialization for UNKNOWN as well.

```
auto elementType = Type::create<TypeKind::UNKNOWN>();
if (!variantArray.empty() && nonNullIndex.has_value()) {
    elementType = variantArray.at(nonNullIndex.value()).inferType();
}
```

Reply:
Thanks for the suggestion, could you please share why the UNKNOWN check can be removed? Would it not be required for cases like variant vectors with all Nulls?

Reply:
`VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL ` will handle UNKNOWN.

Reply:
Got it, thanks!